wagtail/wagtailadmin/views/pages.py

```diff
@@ -629,8 +629,17 @@ def unpublish(request, page_id):
     next_url = get_valid_next_url_from_request(request)
 
     if request.method == 'POST':
+        include_descendants = request.POST.get("include_descendants", False)
+
         page.unpublish()
 
+        if include_descendants == 'on':
+            live_descendant_pages = page.get_descendants().live()
+            for live_descendant_page in live_descendant_pages:
+                live_descendant_page = live_descendant_page.specific
```

Comment:
Consider using `live_descendant_pages = page.get_decendants().live().specific()` instead - it should reduce the number of DB queries doing this on large, diverse trees.